### PR TITLE
146 refactor scheduling service

### PR DESCRIPTION
**Pull request by Saeid-Solei**
# PR Details
## Description

Refactor scheduling_service to use streets_vehicle_list and streets_vehicle_schedule libraries

## Related Issue

#146 

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [x] I have added any new packages to the sonar-scanner.properties file
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: scheduling_service/include/scheduling_worker.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+
+		std::shared_ptr<OpenAPI::OAIIntersection_info> intersection_info;
+
+		
+		void connect_intersection_info();
+
+	public:
+		// scheduling_worker(/* args */);
+		// ~scheduling_worker();
```
Comment (paulbourelly999) [inline scheduling_service/include/scheduling_worker.h:31]: Remove any commented out lines

File: scheduling_service/manifest.json [hunk 2]
```diff
@@ -26,9 +26,15 @@
...
+            "name": "intersection_type",
+            "value": "stop_controlled_intersection",
+            "description": "The type of the intersection, e.g., stop_controlled_intersection, signalized_intersection",
+            "type": "STRING"
+        },
         {
             "name": "exp_delta",
-            "value": 300,
+            "value": 60000,
             "description": "Time(in seconds) without receiving status & intent updates after which a vehicle is no longer considered!",
```
Comment (paulbourelly999) [inline scheduling_service/manifest.json:38]: Update description to be milliseconds right?

File: scheduling_service/src/main.cpp [hunk 3]
```diff
@@ -538,59 +112,38 @@ void call_scheduling_thread(){
...
-                metadata.AddMember("intersection_type", "Carma/stop_controlled_intersection",allocator);
-                document.AddMember("metadata", metadata, allocator);
-
-                Value schedule;
-                if (!list_veh_copy.empty()){
-                    schedule = scheduling_func(list_veh_copy, allocator, last_schedule);
-                }
-                document.AddMember("payload", schedule, allocator);
+                SPDLOG_INFO("schedule number #{0}", sch_count);      
+                auto t = system_clock::now() + milliseconds(int(scheduling_delta*1000));
```
[the remark this answers is not in the document]
Comment (Saeid-Solei) [reply scheduling_service/src/main.cpp:119]: Yes, I believe so.

File: scheduling_service/src/main.cpp [hunk 4]
```diff
@@ -538,59 +108,38 @@ void call_scheduling_thread(){
...
-                StringBuffer buffer;
-                Writer<StringBuffer> writer(buffer);
-                document.Accept(writer);
-                string msg_to_send = buffer.GetString();
+                streets_vehicle_scheduler::intersection_schedule schedule;
+                schedule.timestamp = duration_cast<milliseconds>(system_clock::now().time_since_epoch()).count();
+                unordered_map<std::string, streets_vehicles::vehicle> veh_map = veh_list->get_vehicles();
+                scheduler->schedule_vehicles(veh_map, schedule);
 
+                string msg_to_send = schedule.toCSV();
```
[the remark this answers is not in the document]
Comment (Saeid-Solei) [reply scheduling_service/src/main.cpp]: Please see the changes.

**Review by dan-du-car — APPROVED** [review]

File: streets_utils/streets_vehicle_scheduler/include/intersection_schedule.h [hunk 5]
```diff
@@ -102,9 +104,9 @@ namespace streets_vehicle_scheduler {
         /**
          * @brief Method to write intersection schedule as JSON scheduling message.
          * 
-         * @return rrapidjson::Value& reference 
+         * @return std::string& reference.
          */
-        rapidjson::Value toJson() const;
+        std::string toJsonString() const;
```
Comment (paulbourelly999) [inline streets_utils/streets_vehicle_scheduler/include/intersection_schedule.h:109]: Don't rename the method. Just add the string functionality on to the existing method.

File: streets_utils/streets_vehicle_scheduler/src/intersection_schedule.cpp [hunk 6]
```diff
@@ -40,20 +40,26 @@ namespace streets_vehicle_scheduler {
 	    return schedule_info;
     }
 
-    rapidjson::Value intersection_schedule::toJson() const {
+    std::string intersection_schedule::toJsonString() const {
         
         rapidjson::Document doc;
+        doc.SetArray();
+        rapidjson::Document::AllocatorType &allocator = doc.GetAllocator();
```
Comment (paulbourelly999) [inline streets_utils/streets_vehicle_scheduler/src/intersection_schedule.cpp:47]: This needs to include metadata and everything. Just add onto the toJson method, keep the name and add the metadata information.  I think you should be able to get the timestamp from the intersection_schedule. We may want to rename this struct eventually to all_stop_intersection_schedule.

**Review by paulbourelly999 — APPROVED** [review]

**Pull request merged**